src/utils.js

```diff
@@ -2,16 +2,25 @@
  * @ignore
  */
 export default class {
-    getPerformanceInfo() {
-        let timing = window.performance.timing,
-            interactive = timing.domInteractive - timing.domLoading,
-            dcl = timing.domContentLoadedEventStart - timing.domLoading,
-            complete = timing.domComplete - timing.domLoading;
-        return {
-            ptInteractive: interactive >= 0 ? interactive : undefined,
-            ptDcl: dcl >= 0 ? dcl : undefined,
-            ptComplete: complete >= 0 ? complete : undefined
-        };
+    getPerformanceInfo(targetWindow) {
+        if('navigation' in window[targetWindow].performance && 'timing' in window[targetWindow].performance){
+            let timing = window[targetWindow].performance.timing,
+                nav = window[targetWindow].performance.navigation,
+                interactive = timing.domInteractive - timing.domLoading,
+                dcl = timing.domContentLoadedEventStart - timing.domLoading,
+                complete = timing.domComplete - timing.domLoading,
+                elapsedMs = (new Date()).getTime() - timing.domLoading;
+            return {
+                ptInteractive: interactive >= 0 ? interactive : undefined,
```

Comment:
Naming of these properties looks a bit arbitrary. I think it'd be better there's some consistency, e.g. `tti` from `Time To Interactive`

And `pt` and `nv` prefixes are a little hard to understand easily outside of this module for me. How about using nested shape, or just flattening?

```ts
return {
  timing /*metrics? */ : { ... },
  navigation: { type: ... }
}

// or 

return {
  tti: ...,
  type: ...,
}
```

Reply:
for the naming, ptInteractive is now ptTti but there is no other change for now.
we need to update the log format for Endpoint.
https://github.com/ingestly/ingestly-endpoint/pull/4

Reply:
regarding the second half of the suggestion...

- As of now, SDK uses the returned values directly for generating the payload. There is no process to map the nested returned values and query parameters.
- Fastly's Logging feature does not support the nested data structure so that is why we are building query parameters.